### PR TITLE
add(bam delivery message)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adds support for a bam delivery message.

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/delivery_message/messages/bam_message.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from cg.services.delivery_message.messages.delivery_message import DeliveryMessage
+from cg.services.delivery_message.messages.utils import get_caesar_delivery_path
+from cg.store.models import Case
+
+
+class BamMessage(DeliveryMessage):
+
+    def create_message(self, cases: list[Case]) -> str:
+        delivery_path: str = get_caesar_delivery_path(cases[0])
+        return (
```
Comment (islean) [inline cg/services/delivery_message/messages/bam_message.py:10]: Would we want to immediately include the case names in the delivery message?

**Review by islean — APPROVED** [review]
Looks nice! We should either fix #3846 with regards to BAM deliveries here or mention this message in the PR fixing that issue

File: cg/services/delivery_message/messages/bam_message.py (hunk 1 above)
Comment (Vince-janv) [reply cg/services/delivery_message/messages/bam_message.py:10]: This is a good point. We decided to have one case per sample so we can't assume that all the samples in an order will be delivered at the same time.

Comment (Vince-janv) [reply cg/services/delivery_message/messages/bam_message.py:10]: We ended up naming the cases f"{sample_name}-case" so listing the cases would be a bit ugly. Ideally we should list the sample.name for each sample delivered. What do you guys think?

Comment (ChrOertlin) [reply cg/services/delivery_message/messages/bam_message.py:10]: Sample names make sense to me.

**Pull request merged**